src/components/utils/in-viewport.tsx

```diff
@@ -0,0 +1,32 @@
+import { RefObject, useEffect, useRef, useState, useCallback } from "react";
+
+function useIsInViewport(offset = 0): [boolean, RefObject<HTMLInputElement>] {
```

Comment:
Please consider using one of the many community implementations of this.

Examples:
- https://www.npmjs.com/package/react-cool-inview
- https://www.npmjs.com/package/use-is-in-viewport
- https://www.npmjs.com/package/react-hook-inview

Apart from leveraging other peoples work all these options also use [IntersectionObserverAPI which the recommended approach to handling such things](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). [It is supported by all the browsers that we target](https://caniuse.com/intersectionobserver).